pkg/controller/iam/iamclient/tfiamclient.go

```diff
@@ -685,7 +685,7 @@ func resolveTargetFieldValue(r *krmtotf.Resource, targetField string) (string, e
 		if val, exists, _ := unstructured.NestedString(r.Spec, strings.Split(key, ".")...); exists {
 			return val, nil
 		}
-		if val, exists, _ := unstructured.NestedString(r.Status, strings.Split(key, ".")...); exists {
+		if val, exists, _ := unstructured.NestedString(r.GetStatusOrObservedState(), strings.Split(key, ".")...); exists {
```

Comment:
This is high risk (touches many code paths in the magic machine), so I'm going to assign to 1.115

Reply:
Since this PR can't make it to 1.114, I'm going to allowlist ComputeNetworkFirewallPolicyAssociation.